electron-{bin,source,chromedriver}_35: init at 35.0.3

teutat3s opened the pull request:
https://github.com/electron/electron/releases/tag/v35.0.0
https://github.com/electron/electron/releases/tag/v35.0.1
https://github.com/electron/electron/releases/tag/v35.0.2
https://github.com/electron/electron/releases/tag/v35.0.3
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

teutat3s committed:
electron_35-bin: init at 35.0.3

- Changelog: https://github.com/electron/electron/releases/tag/v35.0.0
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.1
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.2
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.3

teutat3s committed:
electron-chromedriver_35: init at 35.0.3

- Changelog: https://github.com/electron/electron/releases/tag/v35.0.0
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.1
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.2
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.3

teutat3s committed:
electron-source.electron_35: init at 35.0.3

- Changelog: https://github.com/electron/electron/releases/tag/v35.0.0
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.1
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.2
- Changelog: https://github.com/electron/electron/releases/tag/v35.0.3

teutat3s committed:
electron: remove duplicate use_qt flag

We already set it in chromium/common.nix, see 062eb42eba19a38847f16cf52dd3477d8987e593.

teutat3s commented:
Sigh. The source build needs some more work, converting to draft.

TomaSajt commented:
I'm glad I put the header generation before the main build, this way we knew that it failed before building the entire browser.

teutat3s commented:
Tracked down a bit to this commit https://github.com/electron/electron/commit/7a5e9613d2c990752ec99b2748403259526e76e5.

TomaSajt commented:
This is the culprit:
https://github.com/electron/electron/blob/e4bd0cd3dcc57cb6889f845c75952bf999cbf581/script/node/generate_node_headers.py#L34-L39

The directory is not called "Testing" in release builds. They expect an env-var to be set?

TomaSajt commented:
This small change fixes it:
```patch
diff --git a/pkgs/development/tools/electron/common.nix b/pkgs/development/tools/electron/common.nix
index f0d84e9d2692..19777b11b584 100644
--- a/pkgs/development/tools/electron/common.nix
+++ b/pkgs/development/tools/electron/common.nix
@@ -64,6 +64,9 @@ in
     hash = info.chromium_npm_hash;
   };

+  # needed for header generation in electron 35 and above
+  env.ELECTRON_OUT_DIR = "Release";
+
   src = null;

   patches =
```

(though this would rebuild all versions, so we should probably only do this conditionally)

teutat3s committed:
electron: 35+ header generation needs ELECTRON_OUT_DIR

https://github.com/electron/electron/pull/44657
https://github.com/electron/electron/blob/e4bd0cd3dcc57cb6889f845c75952bf999cbf581/script/node/generate_node_headers.py#L34-L39

Co-authored-by: TomaSajt <62384384+TomaSajt@users.noreply.github.com>
Co-authored-by: Emily <git@emilylange.de>

TomaSajt commented:
I highly doubt lib.optionals is what we want, that's for lists.
Maybe just `if cond then val else null` is the easiest?
Not sure if that causes a rebuild or not.

teutat3s commented:
Another nice surprise:
```
[41984/43140] ACTION //third_party/electron_node:generate_config_gypi(//build/toolchain/linux/unbundle:default)
FAILED: gen/third_party/electron_node/config.gypi
python3 ../../third_party/electron_node/tools/generate_config_gypi.py gen/third_party/electron_node/config.gypi --out-dir . --dep-file gen/third_party/electron_node/generate_config_gypi.d --node-gn-path //third_party/electron_node
Traceback (most recent call last):
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 113, in <module>
    main()
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 100, in main
    config = get_gn_config(args.out_dir)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 32, in get_gn_config
    gn_args = subprocess.check_output(
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 468, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 550, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 1028, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 1963, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'gn'
```

TomaSajt commented:
My main suspect upon first glance is `out_dir`. I encountered it when I was looking at where else `ELECTRON_OUT_DIR` was used. I don't have time to track it down today.

It sucks a lot that this was so late in the build process.
Perhaps we can add an extra job (like the header job) to the build so that we don't have to build everything.

teutat3s committed:
chromium: move gnChromium to nativeBuildInputs,

because electron 35+ needs gn in PATH at build time

[41984/43140] ACTION //third_party/electron_node:generate_config_gypi(//build/toolchain/linux/unbundle:default)
FAILED: gen/third_party/electron_node/config.gypi
python3 ../../third_party/electron_node/tools/generate_config_gypi.py gen/third_party/electron_node/config.gypi --out-dir . --dep-file gen/third_party/electron_node/generate_config_gypi.d --node-gn-path //third_party/electron_node
Traceback (most recent call last):
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 113, in <module>
    main()
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 100, in main
    config = get_gn_config(args.out_dir)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 32, in get_gn_config
    gn_args = subprocess.check_output(
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 468, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 550, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 1028, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 1963, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'gn'

TomaSajt commented:
Ideally we should be using `gnChromium` (an override of `gn` that is used to execute the listed `buildTargets`). Unfortunately it is not exposed directly.

IMO ideally, in the `chromium/common.nix` file, `gnChromium` should be put inside `nativeBuildInputs` instead of just string-iterpolating it.

teutat3s commented:
3.5 hours later:

```
FAILED: gen/third_party/electron_node/config.gypi
python3 ../../third_party/electron_node/tools/generate_config_gypi.py gen/third_party/electron_node/config.gypi --out-dir . --dep-file gen/third_party/electron_node/generate_config_gypi.d --node-gn-path //third_party/electron_node
Traceback (most recent call last):
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 113, in <module>
    main()
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 100, in main
    config = get_gn_config(args.out_dir)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 32, in get_gn_config
    gn_args = subprocess.check_output(
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 468, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/26yi95240650jxp5dj78xzch70i1kzlz-python3-3.12.9/lib/python3.12/subprocess.py", line 573, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['gn', 'args', '--list', '--short', '-C', '.']' returned non-zero exit status 1.
```

ljuzig commented:
The Electron build script should include a flag similar to `-Wno-unknown-warning-option`, as already used in the Chromium build script:

```bash
# Don't spam warnings about unknown warning options. This is useful because
# our Clang is always older than Chromium's and the build logs have a size
# of approx. 25 MB without this option (and this saves e.g. 66 %).
env.NIX_CFLAGS_COMPILE = "-Wno-unknown-warning-option";
```

Currently, building Electron generates around 40k lines of warnings like this one:

```
warning: unknown warning option '-Wno-nontrivial-memcall'; did you mean '-Wno-nontrivial-memaccess'? [-Wunknown-warning-option]
```

Adding this flag would significantly reduce unnecessary log verbosity.

ljuzig commented:
The build still fails:

```
FAILED: gen/third_party/electron_node/config.gypi 
python3 ../../third_party/electron_node/tools/generate_config_gypi.py gen/third_party/electron_node/config.gypi --out-dir . --dep-file gen/third_party/electron_node/generate_config_gypi.d --node-gn-path //third_party/electron_node
Traceback (most recent call last):
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 113, in <module>
    main()
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 105, in main
    f.write(repr(translate_config(args.out_dir, config, v8_config)))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 62, in translate_config
    'node_module_version': int(config['node_module_version']),
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '"133"'
```

TomaSajt commented:
It would be very helpful to me/others if you included some extra lines before the actual failure so that I and others can see which build task is failing.

TomaSajt commented:
A quick PSA: you can add this to the front of the build tasks to immediately trigger the failure:
 `third_party/electron_node:generate_config_gypi`

teutat3s commented:
Some python findings that might be obvious to someone with more python kung fu:

```
❯ nix run nixpkgs#python3
Python 3.12.8 (main, Dec  3 2024, 18:42:41) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> int('133')
133
>>> int('"133"')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '"133"'
```

teutat3s commented:
Could this be all that needs to change?

```
diff --git i/pkgs/development/tools/electron/info.json w/pkgs/development/tools/electron/info.json
index 6847c0d38e98..4b03df4101f8 100644
--- i/pkgs/development/tools/electron/info.json
+++ w/pkgs/development/tools/electron/info.json
@@ -1937,7 +1937,7 @@
             }
         },
         "electron_yarn_hash": "0l38rbmlrcrgkw7ggj33xszcs7arm601gzq4c8v0rn3m5zp6yr77",
-        "modules": "133",
+        "modules": 133,
         "node": "22.14.0",
         "version": "35.0.1"
     }
```

ljuzig commented:
> It would be very helpful to me/others if you included some extra lines before the actual failure so that we can see which build task is failing. (I assume it's still the same place as the other failures)
> 
> In any case, I'll try tracking down the issue. I'll get back to you if I found anything.

**"Sorry, but I was already trying to rebuild it before reading this message, and I managed to build it successfully just by forcing this line:** 
```diff
-node_module_version = info.modules;
+node_module_version = lib.toInt info.modules;
```  
[[Line 185 of common.nix](https://github.com/NixOS/nixpkgs/blob/fb25d7111b7d2937554220aab478e39e7a1e93c7/pkgs/development/tools/electron/common.nix#L185)](https://github.com/NixOS/nixpkgs/blob/fb25d7111b7d2937554220aab478e39e7a1e93c7/pkgs/development/tools/electron/common.nix#L185)  

**Of course, my fix is not ideal—I just needed it to build Electron as quickly as possible—but it might help in identifying the issue."**

TomaSajt commented:
> Could this be all that needs to change?
> 
> ```
> diff --git i/pkgs/development/tools/electron/info.json w/pkgs/development/tools/electron/info.json
> index 6847c0d38e98..4b03df4101f8 100644
> --- i/pkgs/development/tools/electron/info.json
> +++ w/pkgs/development/tools/electron/info.json
> @@ -1937,7 +1937,7 @@
>              }
>          },
>          "electron_yarn_hash": "0l38rbmlrcrgkw7ggj33xszcs7arm601gzq4c8v0rn3m5zp6yr77",
> -        "modules": "133",
> +        "modules": 133,
>          "node": "22.14.0",
>          "version": "35.0.1"
>      }
> ```

Perhaps.
I didn't know what exactly was inside `info.json` so I didn't occur to me that it might be the culprit.

If the subtask works, hopefully the entire build works too.

TomaSajt commented:
> > It would be very helpful to me/others if you included some extra lines before the actual failure so that we can see which build task is failing. (I assume it's still the same place as the other failures)
> > In any case, I'll try tracking down the issue. I'll get back to you if I found anything.
> 
> Sorry, but I was already trying to rebuild it before reading this message, and I managed to build it successfully just by forcing this line:
> 
> ```diff
> -node_module_version = info.modules;
> +node_module_version = lib.toInt info.modules;
> ```
> 
> [Line 185 of common.nix](https://github.com/NixOS/nixpkgs/blob/fb25d7111b7d2937554220aab478e39e7a1e93c7/pkgs/development/tools/electron/common.nix#L185)
> 
> Of course, my fix is not ideal—I just needed it to build Electron as quickly as possible—but it might help in identifying the issue.

I assume this might break electron versions below 35. Should we conditionalize?

ljuzig commented:
> I assume this might break electron versions below 35. Should we conditionalize?

I can try to build it with `133` as int in the `info.json` and get back to you.
Also, why `133` and not `134`? Electron 35 shoud have `Chromium v134`.
Running the built electron still get me Chromium v134 tho.

TomaSajt commented on pkgs/development/tools/electron/common.nix:
```diff
@@ -64,6 +64,9 @@ in
   };
 
+  # Needed for header generation in electron 35 and above
+  env.ELECTRON_OUT_DIR = if (lib.versions.major info.version >= "35") then "Release" else null;
```
btw this breaks `electron_33`, since `env.*` can only be a string (forgot about this). It would probably be OK if we just used `optionalString`

teutat3s replied:
Yeah I also found this today in my testing.

ljuzig commented:
> Could this be all that needs to change?
> 
> ```
> diff --git i/pkgs/development/tools/electron/info.json w/pkgs/development/tools/electron/info.json
> index 6847c0d38e98..4b03df4101f8 100644
> --- i/pkgs/development/tools/electron/info.json
> +++ w/pkgs/development/tools/electron/info.json
> @@ -1937,7 +1937,7 @@
>              }
>          },
>          "electron_yarn_hash": "0l38rbmlrcrgkw7ggj33xszcs7arm601gzq4c8v0rn3m5zp6yr77",
> -        "modules": "133",
> +        "modules": 133,
>          "node": "22.14.0",
>          "version": "35.0.1"
>      }
> ```

This leads to this error:

```
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

      [...]

       error: expected a string but found an integer: 133

```

I think I already tried that before ending up with my solution. I guess a straight conversion to int could lead to breaking Electron < 35 as suggested by @TomaSajt.

TomaSajt commented:
> > Could this be all that needs to change?
> > ```
> > diff --git i/pkgs/development/tools/electron/info.json w/pkgs/development/tools/electron/info.json
> > index 6847c0d38e98..4b03df4101f8 100644
> > --- i/pkgs/development/tools/electron/info.json
> > +++ w/pkgs/development/tools/electron/info.json
> > @@ -1937,7 +1937,7 @@
> >              }
> >          },
> >          "electron_yarn_hash": "0l38rbmlrcrgkw7ggj33xszcs7arm601gzq4c8v0rn3m5zp6yr77",
> > -        "modules": "133",
> > +        "modules": 133,
> >          "node": "22.14.0",
> >          "version": "35.0.1"
> >      }
> > ```
> 
> This leads to this error:
> 
> ```
> error:
>        … while calling the 'derivationStrict' builtin
>          at <nix/derivation-internal.nix>:34:12:
>            33|
>            34|   strict = derivationStrict drvAttrs;
>              |            ^
>            35|
>      
>       … while evaluating derivation 'nixos-system-nixos-24.11.715519.ebe2788eafd5'
>          whose name attribute is located at /nix/store/zjwa5665x3izws6abpr4j9cdg4a5q9x6-nixos-24.11/nixos/pkgs/stdenv/generic/make-derivation.nix:336:7
> 
>        … while evaluating attribute 'buildCommand' of derivation 'nixos-system-nixos-24.11.715519.ebe2788eafd5'
>          at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/top-level.nix:1:2272:
>        … while evaluating the option `system.systemBuilderCommands':
> 
>       [...]
> 
>        error: expected a string but found an integer: 133
> ```
> 
> I think I already tried that before ending up with my solution. But I guess a straight conversion to int with `lib.toInt` could lead to breaking Electron < 35 as suggested by @TomaSajt.

How can this possibly be an eval error?
Could you run with `--show-trace`

teutat3s committed:
electron: node_module_version should be int

FAILED: gen/third_party/electron_node/config.gypi
python3 ../../third_party/electron_node/tools/generate_config_gypi.py gen/third_party/electron_node/config.gypi --out-dir . --dep-file gen/third_party/electron_node/generate_config_gypi.d --node-gn-path //third_party/electron_node
Traceback (most recent call last):
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 113, in <module>
    main()
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 105, in main
    f.write(repr(translate_config(args.out_dir, config, v8_config)))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/src/out/Release/../../third_party/electron_node/tools/generate_config_gypi.py", line 62, in translate_config
    'node_module_version': int(config['node_module_version']),
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '"133"'

Co-authored-by: Ljuzig <diamond.bertelli@gmail.com>

TomaSajt commented:
So we either have to do `lib.toInt` in `common.nix` or change "133" to 133 in `info.json`. Doing both leads to the eval error above.

But if we consider that the update script is pretty complicated, we might be better of doing `lib.toInt`.

ljuzig commented:
> But if we consider that the update script is pretty complicated, we might be better of doing `lib.toInt`.
I'm currently rebuilding `electron 33` with `lib.toInt`. I'll update you if the build succeeds.

ljuzig commented:
`electron 33` built without any errors using `lib.toInt`:
```
[nix-shell]$ electron -v
v33.4.2
```

teutat3s commented:
I built `electron_35` successfully from the last commit of this PR.

emilylange commented on pkgs/development/tools/electron/common.nix:
```diff
@@ -182,7 +185,7 @@ in
     is_electron_build = true;
     root_extra_deps = [ "//electron" ];
-    node_module_version = info.modules;
+    node_module_version = lib.toInt info.modules;
```
I find it a bit funny that you opted to use `lib.toInt` here. But not in a discouraging way. Feel free to do whatever floats you boat. Also, I am no longer involved in maintaining electron.
And not wanting to touch the update script feels like a common pattern. Especially because this is returned as a string by the releases API.

Rather, because I find the chain of function invocations funny:

`builtins.fromJSON` reads it as string -> `lib.toInt` -> `if lib.isInt value` -> `toString` (instead of `mkGnString` if it would already be a string):

https://github.com/NixOS/nixpkgs/blob/71d3a69606c4600e7d6b59c0d5af2071d4f01f2c/pkgs/applications/networking/browsers/chromium/common.nix#L130-L151

emilylange reviewed (commented):
Reviewing because @teutat3s pinged me in private regarding the `gnChromium` change and then saw the `env` merging issue.

Not making this a blocking review, even despite the `env` thingy.

teutat3s commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 387294`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>electron-chromedriver_35</li>
    <li>electron_35</li>
    <li>electron_35-bin</li>
  </ul>
</details>

teutat3s commented on pkgs/development/tools/electron/common.nix:
```diff
@@ -64,6 +64,9 @@ in
   };
 
+  # Needed for header generation in electron 35 and above
+  env.ELECTRON_OUT_DIR = lib.optionalString (lib.versions.major info.version >= "35") "Release";
```
Thank you for the review, this looks a lot better.

teutat3s committed:
electron-source: try to avoid log spam

As a precaution and to avoid problems with hydra builds that had to
be fixed in the past. References: ce04dc5 (#378988) and 3f514d3.

teutat3s commented on pkgs/development/tools/electron/common.nix:
```diff
@@ -213,6 +215,11 @@ in
+  # our Clang is always older than Chromium's and the build logs have a size
+  # of approx. 25 MB without this option (and this saves e.g. 66 %).
+  env.NIX_CFLAGS_COMPILE = "-Wno-unknown-warning-option";
+
```
Thanks for the advice, I tried to implement this in https://github.com/NixOS/nixpkgs/pull/387294/commits/c7613769af0af9beb14f7b1bdfb45febac3f1bf9.

teutat3s commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 387294`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>electron-chromedriver_35</li>
    <li>electron_35</li>
    <li>electron_35-bin</li>
  </ul>
</details>

teutat3s commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 387294`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>electron_34</li>
  </ul>
</details>

teutat3s commented:
Bumped to 35.0.3 and fixed conflicts.

TomaSajt reviewed (approved):
LGTM

I do not really have the time to build from source, but I'm assuming it still builds.

teutat3s commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 387294`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>electron-chromedriver_35</li>
    <li>electron_35</li>
    <li>electron_35-bin</li>
  </ul>
</details>

TomaSajt commented:
Thanks, I'll merge soon :)
